docs/tutorials/zero-to-hero.md

```diff
@@ -308,601 +116,78 @@ export function getResponseByKey(key: string): string {
+> Update the smart contract "plumbing" code: in `src/main.js`, update the contract definition. Replace the values of `viewMethods` and `changeMethods` with our new smart contract methods.
 ```js
-async function saveOracleQueryParams() {
-  let url = document.getElementById('url').value
-  let uid = document.getElementById('uid').value
-  let payload = document.getElementById('payload').value
-  let status = document.getElementById("status")
-  // logging for visibility
-  console.log('sending Params to the blockchain')
-  status.innerText = "sending Params to the blockchain"
-
-  await contract.setOracleQueryParams({ url, uid, payload });
-  // logging for visibility
-  console.log('Params saved to the blockchain')
-  status.innerText = "Params saved to the blockchain"
-  setTimeout(() => status.innerText = "", 1500)
-}
+  // Initializing our contract APIs by contract name and configuration.
+  window.contract = await window.near.loadContract(nearConfig.contractName, {
+    // NOTE: This configuration only needed while NEAR is still in development
+    // View methods are read only. They don't modify the state, but usually return some value.
+    viewMethods: ['getResponse'],
+    // Change methods can modify the state. But you don't receive the returned value when called.
+    changeMethods: ['setResponse'],
+    // Sender is the account ID to initialize transactions.
+    sender: window.accountId,
+  });
 ```
-
-Awesome. We now have a front end which allows us to save API params to the blockchain. Easy-peasy right?
-
-Let's check to make sure everything is working by clicking the button `Save API Params to Blockchain`, and then calling `getOracleQueryParams` on the dev console to make sure our params were saved correctly.
-
-> In NEAR Studio
-> - Click **Run** to redeploy the application
->
-> In the application
-> - click **Save API Params to Blockchain**
->
-> In the developer console
-> - Invoke our new contract method using the following line of code
-
+> in `src/index.html`, add the following element `Price of bitcoin: <div id="response"/>` after the button.
+> in `src/main.js`, add the following function to read the data from the blockchain and update the UI. (Note: you can replace the updateWhoSaidHi function with this new function).
 ```js
-await contract.getOracleQueryParams()
-```
-
-You should see the following output from the blockchain printed to the developer console
-
+// Function to update the page data
+function updateData() {
+  // JavaScript tip:
+  // This is another example of how to use promises. Since this function is not async,
+  // we can't await for `contract.getResponse()`, instead we attaching a callback function
+  // using `.then()`.
+  contract.getResponse().then((response) => {
+    const el = document.getElementById('response');
+    el.innerText = response || 'No data available';
+  });
+}
+```
+> in `src/main.js`, update the code that automatically refreshes the page data to call the new function
```

Comment:
The user is not familiar with this codebase, let's just tell them exactly where it is.
Replace:
`setTimeout(updateWhoSaidHi, 1000);`
with
`setTimeout(updateData, 1000);`